src/main/java/io/burt/jmespath/contrib/function/TokenizeFunction.java

```diff
@@ -0,0 +1,32 @@
+package io.burt.jmespath.contrib.function;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import io.burt.jmespath.Adapter;
+import io.burt.jmespath.JmesPathType;
+import io.burt.jmespath.function.ArgumentConstraints;
+import io.burt.jmespath.function.FunctionArgument;
+
+public class TokenizeFunction extends RegularExpressionFunction {
+  public TokenizeFunction() {
+    super(ArgumentConstraints.listOf(1, 3, ArgumentConstraints.typeOf(JmesPathType.STRING)));
+  }
+
+  @Override
+  protected <T> T callFunction(Adapter<T> runtime, List<FunctionArgument<T>> arguments) {
+    List<T> result = new LinkedList<>();
+    Pattern pattern = (arguments.size() > 1) ? getPattern(runtime, arguments) : Pattern.compile("\\s+");
+    for (String parts: pattern.split(getInputString(runtime, arguments), -1)) {
+      /*
+       * if arguments.size == 1 then leading and trailing empty parts has to be discarded
+       * but the pattern above ensures that nowhere else such an empty part may occur.
+       */
```

Comment:
I think that what this comment says is clear enough from the code, but if you don't think so I suggest you refactor the code until you feel that it explains what's going on well enough to not require a comment.

Comments tend to get left unchanged when the code changes, and in my experience mostly cause confusion in the end.